_posts/2023-11-13-bright-manas-partnership.md

```diff
@@ -0,0 +1,31 @@
+---
+title: "Bright and Manas partner together to create Crystal development tools"
+author: beta-ziliani
+summary: "The story behind four development tools that were released recently"
+---
+
+<img src="/assets/sponsors/manas.png" height="200"><img src="/assets/sponsors/bright.png" height="200">
+
+In the recent [Crystal 1.10 release](/2023/10/09/1.10.0-released/) two new compiler tools were introduced, `crystal tool dependencies` and `crystal tool unreachable`. In parallel, the Crystal team also released [perf-tools](https://github.com/crystal-lang/perf-tools), a shard with tools for tracking memory usage and fibers. In this post we delve into the story of how these tools came to be.
+
+The development of these tools was sponsored by [Bright](https://brightsec.com/), maker of an intelligent exploiter for securing websites. The exploiter works by searching for endpoints and attacking them with a large set of potential security threats. As an established product, it has grown organically over the years, requiring special tools to improve it. Therefore, Bright asked Manas to help out in two major subjects: refactoring of the application and memory leak hunting.
+
+## Refactoring
+
+The application has two distinctive features: discovering endpoints, and attacking them. From an architectural point of view, it made sense to split them in two different applications. This was a major refactor that required the construction of `crystal tool dependencies`, which shows a tree of `require` dependencies. It becomes easier to split the application following the require graph.
+
+Additionally, `crystal tool dependencies` helps understanding conflicts arising from the `require` order. For instance, if two files `a.cr` and `b.cr` define the same method for a class, Crystal uses the last definition that comes the last file being required. With this tool we can observe which file is the last one required between `a.cr` and `b.cr`, and understand why.
```

Comment:
**Issue:** The second sentance doesn't seem to make sense. Is there a word missing?

Reply:
How about

```suggestion
Additionally, `crystal tool dependencies` helps understanding conflicts arising from the `require` order. For instance, if two files `a.cr` and `b.cr` define the same method for a class, Crystal uses the definition coming from the last file being required. With this tool we can then observe the order in which `a.cr` and `b.cr` are required and therefore understand why Crystal is calling one of the definitions and not the other.
```

Reply:
That's understandable now 👍